File: 015/problem015.py
#!/usr/bin/python

import math


# n = steps
# k = grid size
def combination(n, k):
    return math.factorial(n) // (math.factorial(k) * (math.factorial(n - k)))


def count_routes(k):
    return combination(2 * k, k)


print count_routes(20)

File: 015/problem015.rb
#!/usr/bin/env ruby
class Integer
  def factorial
    return 1 if zero?
    return 1 if self == 1
    self * (self - 1).factorial
  end
end

def combination(n, k)
  n.factorial / (k.factorial * (n - k).factorial)
end

def count_routes(k)
  combination(2 * k, k)
end

puts count_routes(20)
